test/e2e/external_oidc_test.go

```diff
@@ -0,0 +1,461 @@
+	c, err = tc.updateProxyForIngressCert(ctx)
+	cleanups = append(cleanups, c...)
+	require.NoError(tc.t, err)
+
+	// sync service-ca signing certificate to the KAS nodes as a static resource so that it can be used with --oidc-ca-file
+	c, err = tc.syncServingCA(ctx)
+	cleanups = append(cleanups, c...)
+	require.NoError(tc.t, err)
+
+	// setup kube-apiserver to access the external OIDC directly by modifying its args via UnsupportedConfigOverrides
+	kasOrigRev, c, err := tc.updateKASArgsForOIDC(ctx, kcClient.IssuerURL())
+	cleanups = append(cleanups, c...)
+	require.NoError(tc.t, err)
+
+	// wait for serving CA and KAS args overrides to get rolled out
+	tc.t.Log("will wait for KAS rollout")
+	err = test.WaitForNewKASRollout(tc.t, ctx, tc.operatorConfigClient.OperatorV1().KubeAPIServers(), kasOrigRev)
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func (tc *testClient) updateProxyForIngressCert(ctx context.Context) (cleanups []func(), err error) {
+	tc.t.Log("will copy default-ingress-cert and patch proxy")
+
+	defaultIngressCert, err := tc.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Get(ctx, "default-ingress-cert", metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+
+	cmCopy := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultIngressCert.Name,
+			Namespace: "openshift-config",
+		},
+		Data: defaultIngressCert.Data,
+	}
+
+	_, err = tc.kubeClient.CoreV1().ConfigMaps("openshift-config").Create(ctx, cmCopy, metav1.CreateOptions{})
+	cleanups = append(cleanups, func() {
+		err := tc.kubeClient.CoreV1().ConfigMaps("openshift-config").Delete(ctx, cmCopy.Name, metav1.DeleteOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			tc.t.Fatalf("cleanup failed for configmap '%s/%s': %v", cmCopy.Namespace, cmCopy.Name, err)
+		}
+	})
+	if err != nil {
+		return
+	}
+
+	proxy, err := tc.configClient.ConfigV1().Proxies().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+
+	origTrustedCAName := proxy.Spec.TrustedCA.Name
+	proxy.Spec.TrustedCA.Name = "default-ingress-cert"
+	proxy, err = tc.configClient.ConfigV1().Proxies().Update(ctx, proxy, metav1.UpdateOptions{})
+	cleanups = append(cleanups, func() {
+		proxy.Spec.TrustedCA.Name = origTrustedCAName
+		proxy, err = tc.configClient.ConfigV1().Proxies().Update(ctx, proxy, metav1.UpdateOptions{})
+		if err != nil {
+			tc.t.Fatalf("cleanup failed for proxy '%s': %v", proxy.Name, err)
+		}
+	})
+	if err != nil {
+		return
+	}
+
+	return
+}
+
+func (tc *testClient) syncServingCA(ctx context.Context) (cleanups []func(), err error) {
+	if len(oidcCABundleConfigMap) == 0 || len(oidcCABundleConfigMapNamespace) == 0 {
+		tc.t.Log("no oidc CA defined; will use system CA")
+		return nil, nil
+	}
+
+	tc.t.Log("will sync OIDC ca-bundle to KAS static pod resources")
+
+	signingKey, err := tc.kubeClient.CoreV1().Secrets(oidcCABundleConfigMapNamespace).Get(ctx, oidcCABundleConfigMap, metav1.GetOptions{})
```

Comment:
One nit, the variable **name** `oidcCABundleConfigMap` indicates the resource is a "ConfigMap", but the API function **name** `CoreV1().Secrets(...)` intends a "Secret" : )

Reply:
That's actually not a nit, indeed it's a config map :) I will update the API call. Nice catch :+1:

Reply:
Fixed in 0e62f4e60d8277f43443976f4fefcc98909f8af5